.github/workflows/ci.yaml

```diff
@@ -0,0 +1,109 @@
+name: CI
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+#    branches:
+#      - master
+#      - lsof # just to be able to run all these builds on our own fork
+jobs:
+  with-docker-root:
+    #if: ${{ false }}  # disable for now
+    name: Inside Container centos-zulu-${{matrix.java}} as root
+    runs-on: ubuntu-latest
+    container:
+      image: centos:latest
+    strategy:
+      matrix:
+        java: [ 8, 11 ]
+      fail-fast: false
+      max-parallel: 4
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - uses: actions/setup-java@v2
+        with:
+          distribution: zulu
+          java-version: ${{matrix.java}}
+      - name: Test with Maven
+        run: ./mvnw clean verify -V -B -D"maven.artifact.threads=64" -D"org.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn" -D"disable-toolchain" -D"java.version=${{matrix.java}}" -D"inside-container=true"
+      - name: Upload Unit Test Results
+        if: always()
+        uses: actions/upload-artifact@v2
+        with:
+          name: Unit Test Results (Container ${{matrix.java}}-zulu-centos as root)
+          path: |
+            **/target/surefire-reports/**
+            **/target/rat.txt
+  with-docker-user:
+    #if: ${{ false }}  # disable for now
+    name: Inside Container centos-zulu-${{matrix.java}} as user
+    runs-on: ubuntu-latest
+    container:
+      image: centos:latest
+      options: --user 1001
```

Comment:
May want `1001:1001` (or `1000:1000`), otherwise it runs with gid `root`

Reply:
Are you sure ? This is issued from the doc and also what I saw others doing and what I saw to be recommended.
Also, if the group were still root, then Clifford's PR would have failed because the behavior would have been the same as running inside root in a container.

Reply:
Side note: from what I understand from this option, this is not choosing the permissions to set. Instead this option defines the user to **run as**. The ubuntu-latest image is from what I saw created with a user inside with uid 1001 and gid 1001 and this is pretty common to do that in docker images from what I saw around. So it's not permissions, it's just a run as 1001.

Reply:
```
 $ docker run --rm -it --user 1001 centos id
uid=1001 gid=0(root) groups=0(root)
 $ docker run --rm -it --user 1001:1001 centos id
uid=1001 gid=1001 groups=1001
```

Reply:
Wow! I'm surprised the doc and community forums do not mention that!
So... In that case, why would @cljohnso's PR be successful with `--user 1001` when this user is then still in group 0 ?
Being in group 0 would allow the user to write regardless of the File#setReadonly() flag in theory, right ?

Reply:
My guess that being in the root group only lets you write to files that are `g+w` with the root group.   Usually this is not the case, for example:

`dr-xr-xr-x 2 root root 12288 Sep 15 14:17 /usr/bin`

```
$ docker run --rm -it --user 1001:1001 centos bash -c 'date > /usr/bin/test'
bash: /usr/bin/test: Permission denied                                                          
$ docker run --rm -it --user 1001 centos bash -c 'date > /usr/bin/test'
bash: /usr/bin/test: Permission denied
$ docker run --rm -it --user root centos bash -c 'date > /usr/bin/test'
$ 
```

Reply:
I have updated this PR and main branch with @cljohnso fix runs fine also in now a "real" user mode ;-)
https://github.com/mathieucarbou/terracotta-utilities/actions/runs/1924286992
Thanks :-)

Reply:
> I believe that being in the root group only lets you write to files that are `g+w` with the root group. Usually this is not the case, for example:

That's probably why all the references I saw about running under a non-root user only provide `--user 1001` because in most of the case this would be enough then...

Reply:
The Linux `root` group isn't special in terms of privilege checking -- only the `root` user (UID 0).  `File.setReadOnly()`, when executed on a Linux platform, removes the `w` bit from each of the USER, GROUP, and OTHER permission groups for the file/directory.  Given that the `root` group is _not_ special, absence of the `w` bit in the file/directory's GROUP permission group prevents writing into that file/directory.  Using a UID of anything other than 0 is sufficient.